generation/base.py

```diff
@@ -364,18 +372,25 @@ def pop_jsonpath(self, json_path, default=NO_ARGUMENT, ensure_exists=False) \
         self.delete_jsonpath(json_path, ensure_exists=ensure_exists)
         return data
 
-    def set_jsonpath(self, json_path:str, insert_value:any, must_exist:bool=False):
+    def set_jsonpath(self, json_path:str, insert_value:any, must_exist:bool=False, overwrite:bool=True):
         """
         Sets value at jsonpath location.
 
         Can create a new key if it doesn't exist.
         """
 
+        path_exists = self.jsonpath_exists(json_path)
+
         # If the path must exist, and is missing, we can
         # raise an error by getting the path
-        if must_exist:
+        if not path_exists and must_exist:
             self.get_jsonpath(json_path)
```

Comment:
Can we actually intentionally raise an error here, instead of calling get_jsonpath?

The error would be like: `raise AssetNotFoundError(f"Path {json_path} does not exist.")`